File: merchant-admin/src/api/shop.js
import request from './request'

export function getShopInfo() {
  return request.get('/merchant/shop')
}

export function updateShopInfo(data) {
  return request.put('/merchant/shop', {
    name: data.name,
    logo: data.logo,
    phone: data.phone,
    email: data.email,
    username: data.username,
    password: data.password || undefined,
    address: data.address,
    description: data.description,
    notice: data.notice,
    minPrice: data.minPrice,
    deliveryFee: data.deliveryFee,
    latitude: data.latitude,
    longitude: data.longitude,
    openTime: data.openTime,
    closeTime: data.closeTime
  })
}

export function toggleBusinessStatus(businessStatus) {
  return request.put('/merchant/shop/business-status', { businessStatus })
}

File: backend/src/main/java/com/example/backend/entity/Shop.java
package com.example.backend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("shop")
public class Shop {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String shopNo;
    private String name;
    private String logo;
    private String description;
    private String phone;
    private String email;
    private String address;
    private Double latitude;
    private Double longitude;
    private String openTime;
    private String closeTime;
    private BigDecimal minPrice;
    private BigDecimal deliveryFee;
    private BigDecimal rating;
    private Integer sales;
    private String notice;
    private Integer status;
    private Integer businessStatus;
    @TableField(exist = false)
    private Double distance;
    private String geohash;
    private String auditRemark;
    private String username;
    private String password;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}
